examples/Assertions/Assertions.ipynb

```diff
@@ -0,0 +1,726 @@
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"responseChatCompletions\": [\n",
+      "    {\n",
+      "      \"role\": \"assistant\",\n",
+      "      \"content\": null,\n",
+      "      \"tool_calls\": [\n",
+      "        {\n",
+      "          \"id\": \"call_kd4VN6DojtGxxQCP1MGVPz3T\",\n",
+      "          \"type\": \"function\",\n",
+      "          \"function\": {\n",
+      "            \"name\": \"weather\",\n",
+      "            \"arguments\": \"{\\n  \\\"city\\\": \\\"San Francisco\\\"\\n}\"\n",
+      "          }\n",
+      "        }\n",
+      "      ]\n",
+      "    }\n",
+      "  ]\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "JSON.stringify(await task(\"What's the weather in San Francisco?\"), null, 2);\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"responseChatCompletions\": [\n",
+      "    {\n",
+      "      \"role\": \"assistant\",\n",
+      "      \"content\": \"I'm sorry, but I can't provide you with your bank balance. You will need to check with your bank directly for that information.\"\n",
+      "    }\n",
+      "  ]\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "JSON.stringify(await task(\"What is my bank balance?\"), null, 2);\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"responseChatCompletions\": [\n",
+      "    {\n",
+      "      \"role\": \"assistant\",\n",
+      "      \"content\": \"I need more information to provide you with the weather. Could you please specify the city and the date for which you would like to know the weather?\"\n",
+      "    }\n",
+      "  ]\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "JSON.stringify(await task(\"What is the weather?\"), null, 2);\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scoring outputs\n",
+    "\n",
+    "Validating these cases is subtle. For example, if someone asks \"What is the weather\", the correct answer is to ask for clarification. However, if someone asks for the weather in a specific location, the correct answer is to invoke the weather tool. How do we validate this variety of responses?\n",
+    "\n",
+    "### Using assertions\n",
+    "\n",
+    "Instead of trying to score a specific responses, we'll use a technique called _assertions_ to validate certain criteria about each response. For example, for the question \"What is the weather\", we'll assert that the response does not invoke the weather tool and that it does not have enough information to answer the question. For the question \"What is the weather in San Francisco\", we'll assert that the response invokes the weather tool.\n",
```

Comment:
```suggestion
    "Instead of trying to score a specific response, we'll use a technique called _assertions_ to validate certain criteria about each response. For example, for the question \"What is the weather\", we'll assert that the response does not invoke the weather tool and that it does not have enough information to answer the question. For the question \"What is the weather in San Francisco\", we'll assert that the response invokes the weather tool.\n",
```